lms/djangoapps/course_api/views.py

```diff
@@ -123,6 +125,39 @@ def get_object(self):
         )
 
 
+class CourseListUserThrottle(UserRateThrottle):
+    """Limit the number of requests users can make to the course list API."""
+    # The course list endpoint is likely being inefficient with how it's querying
+    # various parts of the code and can take courseware down, it needs to be rate
+    # limited until optimized. LEARNER-5527
+
+    THROTTLE_RATES = {
+        'user': '2/minute',
+        'staff': '10/minute',
+    }
+
+    def allow_request(self, request, view):
+        if USE_RATE_LIMIT_10_FOR_COURSE_LIST_API.is_enabled():
```

Comment:
@BenjiLee: It would be great if we could encapsulate this in a tool that uses a ConfigModel, or some other functionality to make this easier, even if the rates don't permanently live in the ConfigModel, but we can use admin to slide the rates to arbitrary numbers without hardcoding them all and creating a switch for each one.  The tool could also separate out the `user` and `staff` values, so if one needs to be larger, we aren't forced to bump the other one as well.  I'm not sure if the current error message that goes to splunk lets us know which user role is getting throttled?

Reply:
@waheedahmed: It looks like we didn't have a unit test per switch in @BenjiLee's code either.  I think he just manually tested that none of the switch combos caused a problem.  I think it is fine to take that approach.

Reply:
Adding this as a future feature in the OEP